week13/상현/PG_GPS.java

```diff
@@ -0,0 +1,67 @@
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+/**
+ * 실패...
+ * dp + dfs로 시도
+ */
+class Solution {
+    private static List<Integer>[] trees;
+    private static int[][] dp;
+    public int solution(int n, int m, int[][] edge_list, int k, int[] gps_log) {
+        int answer = 0;
+
+        trees = new ArrayList[n + 1];
+        dp = new int[n + 1][k];
+        for (int i = 1; i <= n; i++) {
+            trees[i] = new ArrayList<>();
+            Arrays.fill(dp[i], Integer.MAX_VALUE - k);
+        }
+
+        for (int i = 0; i < m; i++) {
+            trees[edge_list[i][0]].add(edge_list[i][1]);
+            trees[edge_list[i][1]].add(edge_list[i][0]);
+        }
+
+        answer = func(gps_log[0], 0,gps_log, k);
+
+        if (answer == Integer.MAX_VALUE - k) answer = -1;
```

Comment:
```suggestion
        if (answer == Integer.MAX_VALUE) answer = -1;
```
아래 제가 드린 코드로 하면
계산 후에도 도달할 수 없는 조건을 Integer.MAX_VALUE로 체크해서 제안 드립니다!

Reply:
오...근데 저거 풀이 보고 푼건데 이해 안되서...일단 for문 dp 보고 dfs로 바꿔보던가 해야할듯...